src/PackageVersions/Installer.php

```diff
@@ -165,7 +165,12 @@ private static function getVersions(Locker $locker, RootPackageInterface $rootPa
         }
 
         foreach ($rootPackage->getReplaces() as $replace) {
-            yield $replace->getTarget() => $replace->getPrettyConstraint() . '@' . $rootPackage->getSourceReference();
+            $version = $replace->getPrettyConstraint();
+            if ($version === 'self.version') {
```

Comment:
Is there something other than `self.version` that we can add to the tests to demonstrate alternate outcomes? Or is this the only "special" version type in `composer.json`?